Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm

```diff
@@ -783,7 +783,15 @@ static String mutationObserverNotificationString()
     context.translate(-m_scrollOffset.width(), -m_scrollOffset.height());
 
     FloatRect clipRect { FloatPoint(m_scrollOffset), size() };
+
     context.clip(clipRect);
+    context.fillRect(clipRect, WebCore::roundAndClampToSRGBALossy([WebCore::CocoaColor grayColor].CGColor));
+    context.scale(m_scaleFactor);
+
+    auto paddingForCentering = centeringOffset();
+    context.translate(paddingForCentering.width(), paddingForCentering.height());
+
+    clipRect.scale(1.0f / m_scaleFactor);
```

Comment:
Should this instead use the coordinate transform function (convertDown...) to make it clear what it's doing?

Reply:
Hmm I attempted that but it seems like the adjustment for the centering offset in there throws off the snapshot and content that gets scrolled into the view doesn't immediately appear in the snapshot unless you scroll further